include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -266,13 +266,19 @@ ORT_API_STATUS(OrtSetSessionThreadPoolSize, _Inout_ OrtSessionOptions* options,
   * If none are called Ort will use its internal CPU execution provider.
   */
 
-ORT_API_STATUS(OrtSessionGetInputCount, _In_ const OrtSession* sess, _Out_ size_t* out);
+typedef enum ONNXSessionInputInclusion {
+  kInputsOnly = 0, /* Inputs only */
+  kIncludeInitializers = 1 /* Inputs with initializers */
+} ONNXSessionInputInclusion;
+
+ORT_API_STATUS(OrtSessionGetInputCount, _In_ const OrtSession* sess, ONNXSessionInputInclusion inclusion, _Out_ size_t* out);
 ORT_API_STATUS(OrtSessionGetOutputCount, _In_ const OrtSession* sess, _Out_ size_t* out);
```

Comment:
Are there any concerns about breaking this API by adding a new required parameter?

Reply:
I talked to people and there is not much concern. I'd prefer not to change it at all. There is a separate effort going on to refactor the whole API as interfaces, so not much concern.

---
In reply to: [325016132](https://github.com/microsoft/onnxruntime/pull/1854#discussion_r325016132) [](ancestors = 325016132)

Reply:
Since we're still not at 1.0, it's ok to break the API if it makes the API better. We won't have a lot of liberty after 1.0.